Fix unit tests for AGL

mkorniichuk opened the pull request:
This PR is **ready** for review.

### Summary
Most of the unit tests are failed with segfault on AGL. The issue is in gmock. The bug is already known and old enoug (https://bugs.launchpad.net/ubuntu/+source/google-mock/+bug/1609793).

NOTE: Current PR depends on https://github.com/Ford-Luxoft/sdl_core/pull/13

### Changelog
* Gmock segfault fix for AGL

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

mkorniichuk committed:
Fix SEGFAULT for unit tests

mkorniichuk committed:
Remove unnecessary UT

Shared library test is not required since it does not test the functionality of SDL

mkorniichuk committed:
Fix cotire for Linux

install step for cotired tests has been fixed

LuxoftAKutsan commented on tools/cmake/helpers/sources.cmake:
```diff
@@ -150,33 +150,29 @@ function(create_test NAME SOURCES LIBS)
     ${SOURCES}
   )
 
+  # TODO: Fix problems with Cotire on Windows and Qt APPLINK-28060
```
@mkorniichuk  avoid links to internal jira in github

LuxoftAKutsan commented on tools/cmake/helpers/sources.cmake:
```diff
@@ -150,33 +150,29 @@ function(create_test NAME SOURCES LIBS)
-    EXCLUDE_FROM_ALL 0
-  )
   add_test(NAME ${NAME}
-  COMMAND ${NAME} --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/)
```
Spaces, Check spaces

LuxoftAKutsan commented:
Changes in gtest? Are you sure?

ZhdanovP commented on src/3rd_party-static/gmock-1.7.0/include/gmock/gmock-spec-builders.h:
```diff
@@ -1399,7 +1399,8 @@ class ActionResultHolder : public UntypedActionResultHolderBase {
  public:
-  void GetValueAndDelete() const { delete this; }
+ explicit ActionResultHolder() {}
+   void GetValueAndDelete() const { delete this; }
```
@mkorniichuk Problems with indent level, isn't it?

mkorniichuk committed:
New gmock version

gtest-1.7 src removed from sdl
gtest-1.8 added instead (downloading from GitHub)

BSolonenko commented on src/3rd_party-static/CMakeLists.GTest:
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
```
@mkorniichuk may be 2.8.11?

BSolonenko replied:
@mkorniichuk  please add Copyright

BSolonenko replied:
@mkorniichuk I'm not sure we need this specifier in this file.

mkorniichuk replied:
@BSolonenko this cmake list will be triggered as a standalone project, so the **cmake_minimum_required** is a required statement. BTW, in such a way, we specify the minimum cmake version supported by gtest.

mkorniichuk committed:
Fix according to comments

Copyright appended
fixed indention in source.cmake
